lib/Model/Query.js

```diff
@@ -293,7 +293,12 @@ Query.prototype._shareSubscribe = function(options, cb) {
 Query.prototype._removeMapIds = function(ids) {
   for (var i = ids.length; i--;) {
     var id = ids[i];
-    delete this.idMap[id];
+    if (this.idMap[id] > 0) {
+      this.idMap[id]--;
+    }
+    if (this.idMap[id] === 0) {
+      delete this.idMap[id];
+    }
```

Comment:
The two are not quite the same.

In your proposal, any ids that hit zero wouldn't get cleaned up in the map unless `_removeMapIds` happened to get called again with one of those zero-ed out ids.